lib/auth/auth_with_roles.go

```diff
@@ -6423,88 +6423,47 @@ func (a *ServerWithRoles) WatchPendingHeadlessAuthentications(ctx context.Contex
 
 // CreateAssistantConversation creates a new conversation entry in the backend.
 func (a *ServerWithRoles) CreateAssistantConversation(ctx context.Context, req *assist.CreateAssistantConversationRequest) (*assist.CreateAssistantConversationResponse, error) {
-	if err := a.action(apidefaults.Namespace, types.KindAssistant, types.VerbCreate); err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	return a.authServer.CreateAssistantConversation(ctx, req)
+	return nil, trace.NotImplemented("CreateAssistantConversation must not be called on auth.ServerWithRoles")
```

Comment:
Can you help me understand why we're breaking from convention and moving RBAC enforcement out of the `with_roles` file?

Reply:
This is the convention shared by the newer, separate gRPC services we've added, such as the trust service. My understanding is that we are trying to move away from a monolithic approach to services.

@rosstimothy @codingllama do you have any more context to add here?

I'm planning on carrying this new approach over to new services that I add in my work for https://github.com/gravitational/teleport/pull/27588, so it'd be good if we agreed on it.

Reply:
Because `auth_with_roles` is only being used when the service is implemented as a part of AuthService https://github.com/gravitational/teleport/blob/f0756bdd656b34a9eeb3ad6d6f7618e0de27aa5b/api/proto/teleport/legacy/client/proto/authservice.proto#L2231
New services, such as Okta, DeviceTrust, Assist, etc are implemented as separate GRPC services. In this case, `auth_wit_roles` is not being used, and each service needs to implement the RBAC separately. Here is another example https://github.com/gravitational/teleport/blob/c8fdd9ce93fc7cd93b00c0453a27f6754686ea5f/lib/auth/auth_with_roles.go#L5106
Does it make sense?

Reply:
Yep, thanks!